draft-ietf-wimse-s2s-protocol.md

```diff
@@ -401,11 +401,17 @@ To validate the WPT in the request, the recipient MUST ensure the following:
 * Optionally, check that the value of the `jti` claim has not been used before in the time window in which the
  respective WPT would be considered valid.
 * If presented in conjunction with an OAuth access token, the value of the `ath` claim matches the hash of that token's value.
-* If presented in conjunction with a Txn-Token, the value of the `tth` claim matches the hash of that token's value.
-* If presented in conjunction with a token conveying end-user identity or authorization context, the value of
- the `oth` claim matches the hash of that token's value.
+* If `Workload-Proof-Token` contains `signed_headers` claim:
+   * The request contains a header for each claim in the `signed_headers`.
+   * Each corresponding value in the claim matches the hash of the value in the header.
+   * Headers not present in the `Workload-Proof-Token` are accepted without further proof.
```

Comment:
Huh? An attacker can add a header with the same name as an existing header and (depending on implementation) get it accepted.

I suggest to remove this bullet and change the first one to:
> The request contains exactly one header for each claim in `signed_headers` and no additional headers.

Reply:
I must admit to being pretty confused at this stage.